apps/aeoracle/src/aeo_state_tree.erl

```diff
@@ -0,0 +1,166 @@
+%%%-------------------------------------------------------------------
+%%% @copyright (C) 2017, Aeternity Anstalt
+%%% @doc
+%%% ADT for keeping the state of oracles
+%%% @end
+%%%-------------------------------------------------------------------
+
+-module(aeo_state_tree).
+
+%% API
+-export([ get_interaction/2
+        , get_oracle/2
+        , empty/0
+        , enter_interaction/2
+        , enter_oracle/2
+        , lookup_interaction/2
+        , lookup_oracle/2
+        , prune/2
+        , root_hash/1
+        ]).
+
+-ifdef(TEST).
+-export([ interaction_list/1
+        , oracle_list/1
+        ]).
+-endif.
+
+%%%===================================================================
+%%% Types
+%%%===================================================================
+
+-type mtree() :: aeu_mtrees:tree(binary(), binary()).
```

Comment:
It would be nice putting more descriptive types than `binary()` for key and value, ideally with type but at least with a var name e.g. `OracleEnabledAccountPubKey :: binary()` and similarly for the value.

Reply:
Adding some proper types...

The var name would really be OracleEnabledAccountPubKeyOrOracleInteractionId which isn't very practical ;-)